### PR TITLE
#21906: Added float32 support ttnn.softmax for dim != rank-1

### LINKED ISSUE #21906: Softmax with dim different then rank - 1 won't work in different data types
### Describe the bug

Softmax won't work with f32 when dim != rank - 1

```py
import torch
import ttnn

from tests.ttnn.utils_for_testing import assert_with_pcc

device_id = 0
device = ttnn.open_device(device_id=device_id)

def create_tensor(shape, device, dtype): 
    torch_tensor = torch.rand(shape, dtype=torch.float32)
    ttnn_tensor = ttnn.from_torch(torch_tensor, layout=ttnn.TILE_LAYOUT, device=device, dtype = dtype)
    return (torch_tensor, ttnn_tensor)

shape = (32, 32)
torch_tensor, ttnn_tensor = create_tensor(shape, device, ttnn.float32)
torch.softmax(torch_tensor, dim = 0)

ttnn.close_device(device)
```

I get this error:
```
/home/mtopalovic/src/tt-metal/ttnn/cpp/ttnn/operations/moreh/moreh_softmax/device/moreh_softmax_device_operation.cpp:94: input.get_dtype() == DataType::BFLOAT16 || input.get_dtype() == DataType::BFLOAT8_B
```

I'm on c564abca3493cf0105f8b9f595477f13e83a8033 which is latest main as of today.

### Steps to reproduce the issue

_No response_

### Expected behavior

_No response_

### Please complete the following environment information

_No response_

### PR DESCRIPTION
**Pull request by mgajewskiTT**
### Ticket
#21906

### Problem description
When using `ttnn.softmax` with `float32` dtype and `dim != rank -1` it produced wrong results. 

### What's changed
- Added `float32` support, 
- Added test for different types. 

### Next steps
As part of the next steps, the softmax will be cleaned up. The Moreh implementation will be merged with the one in the normalization directory, and the tests will be unified. [#26841](https://github.com/tenstorrent/tt-metal/issues/26841)

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes https://github.com/tenstorrent/tt-metal/actions/runs/16961652510
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/16933458487
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-quick-trigger.yaml) CI passes (if applicable)
- [ ] [TG demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/moreh/moreh_softmax/device/softmax_w_large/softmax_w_large.cpp [hunk 1]
```diff
@@ -42,6 +43,10 @@ MorehSoftmaxOperation::MorehSoftmaxWLargeFactory::create(
     auto [math_fidelity, math_approx_mode, fp32_dest_acc_en, packer_l1_acc, dst_full_sync_en] =
         get_compute_kernel_config_args(arch, compute_kernel_config);
 
+    if (input.dtype() == DataType::FLOAT32) {
+        fp32_dest_acc_en = true;
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/moreh/moreh_softmax/device/softmax_w_large/softmax_w_large.cpp]: Where is the compute kernel config being set up?
Is it [potentially] passed in from the user?
If yes, then we should either not overwrite the user's decision, or print out a warning.
That will require knowing whether it's the defaults or user provided.

Comment (mgajewskiTT) [reply ttnn/cpp/ttnn/operations/moreh/moreh_softmax/device/softmax_w_large/softmax_w_large.cpp]: It can be set by the user, but in all our cases it defaults to the value for the given architecture. A better approach might be to add a flag indicating whether it was explicitly set by the user. If it was set, we should throw an error since reduce LLK won’t work(if it's not true when data is float32). If it wasn’t set by the user, we can safely assign the appropriate value here. I'll work on that.

Comment (mgajewskiTT) [reply ttnn/cpp/ttnn/operations/moreh/moreh_softmax/device/softmax_w_large/softmax_w_large.cpp]: I updated it one layer higher, where we create default compute kernel config and left only check in program factory.

**Review by aczajkowskiTT — APPROVED** [review]
@mgajewskiTT based on conversations, lets also sync up with the training team about their softmax version

**Review by bbradelTT — APPROVED** [review]
